src/client/src/orangehrmCorporateDirectoryPlugin/components/EmployeeDetails.vue

```diff
@@ -0,0 +1,224 @@
+  <oxd-divider></oxd-divider>
+  <div
+    class="orangehrm-directory-card-hover"
+    @mouseleave="showEmailClip = false"
+    @mouseover="showEmailClip = true"
+  >
+    <div class="orangehrm-directory-card-hover-body">
+      <oxd-text type="toast-message">{{ $t('Work Email') }}</oxd-text>
+      <oxd-text ref="cloneEmail" type="toast-title">
+        {{ employeeWorkEmail }}
+      </oxd-text>
+    </div>
+    <div
+      class="orangehrm-directory-card-hover-body orangehrm-directory-card-icon"
+    >
+      <oxd-icon-button
+        v-show="showEmailClip"
+        name="clipboard-check"
+        @click="copyEmail"
+      ></oxd-icon-button>
+    </div>
+  </div>
+  <oxd-divider></oxd-divider>
+</template>
+
+<script>
+import OxdDivider from '@ohrm/oxd/core/components/Divider/Divider';
+import {APIService} from '@/core/util/services/api.service';
+
+export default {
+  name: 'EmployeeDetails',
+  components: {
+    'oxd-divider': OxdDivider,
+  },
+  props: {
+    employeeId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup() {
+    const http = new APIService(
+      'https://07bd2c2f-bd2b-4a9f-97c7-cb744a96e0f8.mock.pstmn.io',
+      'api/v2/corporate-directory/employees',
+    );
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      employeeWorkTelephone: null,
+      employeeWorkEmail: null,
+      showTelephoneClip: false,
+      showEmailClip: false,
+    };
+  },
+  beforeMount() {
+    this.http.get(this.employeeId).then(response => {
+      const {data} = response.data;
+      this.employeeWorkTelephone = data.contactInfo?.workTelephone;
+      this.employeeWorkEmail = data.contactInfo?.workEmail;
+    });
+  },
+  methods: {
+    copyEmail() {
+      navigator.clipboard?.writeText(this.employeeWorkEmail);
+    },
+    copyTelephone() {
+      navigator.clipboard?.writeText(this.employeeWorkTelephone);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.orangehrm-directory-card {
+  &-rounded-body {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    padding-right: 1rem;
+    padding-left: 1rem;
+    margin-top: 1rem;
+    margin-bottom: 1rem;
+    border-radius: 100px;
+    width: 140px;
+    height: 64px;
+    box-shadow: 5px 5px 5px 5px #fafafc;
+    margin-right: 8px;
+  }
+
+  &-hover {
+    display: flex;
+    justify-content: space-between;
+    padding: 0.25rem 0.5rem;
+    border-radius: 0.5rem;
+    width: 156px;
+    min-height: 48px;
+    margin-right: 8px;
+
+    &-body {
+      display: block;
+      align-items: center;
+      overflow: hidden;
+      word-wrap: break-word;
+    }
+  }
+
+  &-hover:hover {
+    background-color: #fafafc;
+  }
+
+  &-icon {
+    margin-right: 0.5rem;
+    color: #64728c;
+    font-size: 24px;
+    display: flex;
+    justify-content: center;
+  }
+}
+
+@media (max-width: 600px) {
+  .orangehrm-directory-card {
+    padding: 0.5rem 1rem;
+    height: auto;
+    overflow: hidden;
+
+    &-rounded-body {
```

Comment:
same styles repeating ?? if the styles needs to be different based on screen size, use media query and define specifically that style. don't repeat everything. For example consider following scenario.

Element with class **selector** needs to have red color font when seeing in mobile screens but green color font when seen from desktop screens.

### BAD PATTERN

```
@include oxd-respond-to('xs') {
  .selector {
    color: red;
    font-size: 12px;
    font-weight: 400;
    margin: 5px;
  }
}

@include oxd-respond-to('md') {
  .selector {
    color: green;
    font-size: 12px;
    font-weight: 400;
    margin: 5px;
  }
}
```

### GOOD PATTERN

```
.selector {
  color: red;
  font-size: 12px;
  font-weight: 400;
  margin: 5px;
  @include oxd-respond-to('md') {
    color: green;
  }
}
```

Reply:
Done